### PR TITLE
feat: Fixup Friends view.

### PR DESCRIPTION
**Pull request by Tejpbit**

### CODE REVIEW

File: android/app/src/main/java/dev/fredag/cheerwithme/ErrorSnackbar.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                    Text(
+                        text = "Ok",
+                    )
+                }
+            }
+        ) {
+            Text(message)
+        }
+    }
+}
```
Comment (Kalior) [inline android/app/src/main/java/dev/fredag/cheerwithme/ErrorSnackbar.kt]:
```suggestion
}

```

File: android/app/src/main/java/dev/fredag/cheerwithme/data/FriendsRepository.kt [hunk 2]
```diff
@@ -3,15 +3,31 @@ package dev.fredag.cheerwithme.data
...
+
 class FriendsRepository @Inject constructor(
     private val backendService: BackendService
 ) {
-    fun getFriends(): Flow<UserFriends> = flow {
-        val friends = backendService.friends()
-        emit(friends)
+    fun getFriends(): Flow<Result<UserFriends>> = flow {
+        try {
+            val request = backendService.friends()
```
Comment (Dr-Horv) [inline android/app/src/main/java/dev/fredag/cheerwithme/data/FriendsRepository.kt:18]: Could we move the body of this try to another method? quite a lot of indentation here

File: android/app/src/main/java/dev/fredag/cheerwithme/data/backend/BackendModule.kt [hunk 3]
```diff
@@ -54,26 +56,30 @@ object BackendModule {
...
             .build()
     }
 
     @Provides
     @Singleton
-    fun backendService(retrofit: Retrofit): BackendService = retrofit.create(BackendService::class.java)
+    fun backendService(retrofit: Retrofit): BackendService =
+        retrofit.create(BackendService::class.java)
 
 }
```
Comment (Kalior) [inline android/app/src/main/java/dev/fredag/cheerwithme/data/backend/BackendModule.kt:85]:
```suggestion
}

```

File: android/app/src/main/java/dev/fredag/cheerwithme/data/backend/BackendModule.kt [hunk 4]
```diff
@@ -54,26 +56,30 @@ object BackendModule {
...
+                it.request().newBuilder().addHeader("Authorization", "Bearer $accessKey").build()
             it.proceed(request)
         }
 
         val client = okHttpClientBuilder.build()
 
         return Retrofit.Builder()
-            .baseUrl("http://192.168.0.113:8080/")
+            //.baseUrl("http://10.0.2.2:8080/")
+            .baseUrl("https://cheerwithme.fredag.dev/")
```
Comment (Dr-Horv) [inline android/app/src/main/java/dev/fredag/cheerwithme/data/backend/BackendModule.kt:74]: :heart_eyes:

**Comment by Tejpbit** [comment]
image

image

**Review by Kalior — COMMENTED** [review]
LGTM

File: android/app/src/main/java/dev/fredag/cheerwithme/friends/FriendsScreen.kt [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+                        ?.let {
+                            item {
+                                Text(text = "Friend Requests".toUpperCase(Locale.getDefault()))
+                            }
+                            it.map {
+                                item {
+                                    FriendRequestItem(
+                                        user = it,
+                                        onAcceptFriendRequest = friendsViewModel::acceptFriendRequest,
+                                        onDenyFriendRequest = {}
```
Comment (Dr-Horv) [inline android/app/src/main/java/dev/fredag/cheerwithme/friends/FriendsScreen.kt]: Add TODO-comment + Issue in Github for this?

**Review by Dr-Horv — APPROVED** [review]

**Pull request merged**